cogboard-app/src/test/kotlin/com/cognifide/cogboard/config/storage/VolumeStorageTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.cognifide.cogboard.config.storage
+
+import com.cognifide.cogboard.config.ConfigType
+import com.cognifide.cogboard.storage.VolumeStorage
+import io.vertx.core.json.JsonObject
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.TestFactory
+import java.io.File
+import java.util.stream.Stream
+
+internal class VolumeStorageTest {
+
+    private val storagePath = VolumeStorageTest::class.java.getResource("/board").path
+
+    private val configs = listOf(
+            Triple(ConfigType.ADMINS, "admins.json", "admins"),
+            Triple(ConfigType.CREDENTIALS, "credentials.json", "credentials"),
+            Triple(ConfigType.BOARDS, "config.json", "boards"),
+            Triple(ConfigType.ENDPOINTS, "endpoints.json", "endpoints")
+    )
+
+    @TestFactory
+    fun `Expect each config file created when not present`(): Stream<DynamicTest> {
+        return configs.stream().map { config ->
+            val type = config.first
+            val fileName = config.second
+            val expectedProp = config.third
+
+            DynamicTest.dynamicTest("Expect $fileName config is created") {
```

Comment:
wow. didnt know that. I used Spock..which is in my opinion...still better ;)

Reply:
Spock has to be used in JUnit 4 in JUnit 5 we have this OOTB